src/riak_kv_backend.erl

```diff
@@ -31,33 +30,52 @@
 -type fold_objects_fun() :: fun((binary(), binary(), term(), any()) ->
                                        any() |
                                        no_return()).
+
+-type index_spec() :: {add, Index, SecondaryKey} | {remove, Index, SecondaryKey}.
```

Comment:
If you just use a variable name like that (i.e. capitalized with no type arguments), it's the same as not specifying any type. If we wanted to be explicit, we could edit it to look like this, which would be functionally identical:
```
-type index_spec() :: {add, Index :: any(), SecondaryKey :: any()} | {remove, Index :: any(), SecondaryKey :: any()}.
```
If I were writing the type spec from scratch, I probably would have gone in and figured out what the actual types are here, instead of just using untyped placeholders like that. However this type spec already existed prior to this PR, but before this, it was copied and pasted manually to every single backend's source file. I didn't take the time to improve the existing spec, but I did at least get rid of all the identical copies scattered around the source tree.